src/layout/mod.rs

```diff
@@ -3595,7 +3595,9 @@ impl<W: LayoutElement> Layout<W> {
                 return;
             };
 
-            let workspace_idx = workspace_idx.unwrap_or(monitors[new_idx].active_workspace_idx);
+            let workspace_idx = workspace_idx.map_or(monitors[new_idx].active_workspace_idx, |v| {
```

Comment:
I think it would look better if you first unwrap_or() then min() with len - 1 (clamp is unnecessary)